easier_net/easier_net.py

```diff
@@ -0,0 +1,174 @@
+        self.num_batches=num_batches
+        self.model_fit_params_file=model_fit_params_file
+        self.score_criterion = (
+            nn.CrossEntropyLoss() if self.num_classes >= 2 else nn.MSELoss()
+        )       
+        if self.model_fit_params_file is not None:
+            self.estimators: List[sier_net.SierNetEstimator] = self.load_model(model_fit_params_file) 
+        else:
+            self.estimators: List[sier_net.SierNetEstimator] = self._generate_estimators()
+
+        assert num_classes != 1
+        assert input_filter_layer
+    
+    def _generate_estimators(self):
+        estimators=[]
+        for est in range(self.n_estimators): #TODO: i -> est?
+            est=sier_net.SierNetEstimator(
+                num_classes=self.num_classes,
+                input_filter_layer=self.input_filter_layer,
+                n_hidden= self.n_hidden,
+                max_iters=self.max_iters,
+                max_prox_iters=self.max_prox_iters,
+                full_tree_pen=self.full_tree_pen,
+                input_pen=self.input_pen,
+            ) 
+            
+            estimators.append(est)
+
+        return estimators
+
+
+    def write_model(self, filename): #save and write model to file
+        #save as .pt file
+        meta_state_dict = self.get_params()
+        meta_state_dict["state_dicts"] = [
+            est.net.state_dict() for est in self.estimators
+        ]
+        torch.save(meta_state_dict, filename)
+
+    def load_model(self, model_fit_params_file):
+        estimators = []
+        state_dict_list = torch.load(model_fit_params_file, map_location='cpu') #to avoid gpu ram surge
+        for state_dict in state_dict_list:
+            sier_net = sier_net.SierNetEstimator(
+                n_inputs=self.n_inputs,
+                num_classes=self.num_classes,
+                input_filter_layer=self.input_filter_layer,
+                n_hidden= self.n_hidden,
+                max_iters=self.max_iters,
+                max_prox_iters=self.max_prox_iters,
+                full_tree_pen=self.full_tree_pen,
+                input_pen=self.input_pen,
+            )
+            sier_net.load_state_dict(state_dict)
+            estimators.append(sier_net)
+        return estimators
+
+    def fit(
+        self,
+        x: np.ndarray,
+        y: np.ndarray,
+    ):
+        self.n_inputs=x.shape[1]
+        for est in self.estimators:
+            est.fit(x, y)
+       
+        return self 
+
+    def score(self, x: np.ndarray, y: np.ndarray) -> float:
+        self.fit(x,y)
+        y_pred = self.predict(x)
+        nparr_to_torch = lambda nparr: (
+            torch.Tensor(nparr)
+            if self.num_classes == 0
+            else torch.from_numpy(nparr.astype(int))[:, 0]
+        )
+        torch_y_pred = nparr_to_torch(y_pred)
+        torch_y = nparr_to_torch(y)
+
+        return -self.score_criterion(torch_y_pred, torch_y).item()
+
+    def predict(self, x: np.ndarray) -> np.ndarray: 
```

Comment:
Did we add the function `predict_proba`? We'll need to add it eventually, but you can do that after we get this initial stuff working